app/src/main/java/com/example/habitsmasher/ui/dashboard/DashboardFragment.java

```diff
@@ -16,13 +17,23 @@
 import com.example.habitsmasher.Habit;
 import com.example.habitsmasher.HabitList;
 import com.example.habitsmasher.R;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
 
 import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
 
-public class DashboardFragment extends Fragment {
+public class DashboardFragment extends Fragment implements HabitDialogListener{
+
+    private static final String TAG = "DashboardFragment";
```

Comment:
This is another minor thing, constants and fields should have a **new line** separating the two.
```
MY_CONSTANT = "CONSTANT"

_myField = "myField"
```